actions/channel_actions.jsx

```diff
@@ -368,12 +368,13 @@ export async function getChannelMembersForUserIds(channelId, userIds, success, e
 }
 
 export async function leaveChannel(channelId, success) {
+    const townsquare = ChannelStore.getByName('town-square');
+    await ChannelActions.selectChannel(townsquare.id);
     await ChannelActions.leaveChannel(channelId)(dispatch, getState);
     if (ChannelUtils.isFavoriteChannelId(channelId)) {
         unmarkFavorite(channelId);
     }
 
-    const townsquare = ChannelStore.getByName('town-square');
     browserHistory.push(TeamStore.getCurrentTeamRelativeUrl() + '/channels/' + townsquare.name);
```

Comment:
This is the bit that's actually doing the channel changing. It will essentially push us through [this react-router route](https://github.com/mattermost/mattermost-webapp/blob/master/routes/route_team.jsx#L298) which you can follow down the rabbit-hole [starting here](https://github.com/mattermost/mattermost-webapp/blob/master/routes/route_team.jsx#L36) if you want to see what's happening. (Warning it's had a lot of organic growth and is in need of cleaning).

Anyways, point is it might work better if you moved the `browserHistory.push()` to be before the `ChannelActions.leaveChannel()` instead of adding the `ChannelActions.selectChannel()` because the `browserHistory.push()` will eventually do the channel select somewhere down the rabbit hole

Let me know if that doesn't make sense and I'll clarify

Reply:
I doubted you, but was wrong. Great tip. It works now on localhost